app/bundles/AssetBundle/Controller/PublicController.php

```diff
@@ -1,92 +1,140 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Mautic\AssetBundle\Controller;
 
-use Mautic\CoreBundle\Controller\FormController as CommonFormController;
+use Doctrine\ORM\EntityNotFoundException;
+use Doctrine\ORM\NonUniqueResultException;
+use Doctrine\ORM\ORMException;
+use Mautic\AssetBundle\Entity\Asset;
+use Mautic\AssetBundle\Model\AssetModel;
+use Mautic\CoreBundle\Controller\AbstractFormController;
 use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Symfony\Component\HttpFoundation\File\Exception\FileNotFoundException;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
-class PublicController extends CommonFormController
+class PublicController extends AbstractFormController
 {
     /**
-     * @param string $slug
+     * Handles public download of assets by slug.
+     *
+     * This method performs the initial validation of the slug, retrieves the
+     *  corresponding Asset entity, and then delegates further response logic
+     *  (such as redirects, streaming, or access denial) to `createAssetResponse()`.
+     *
+     * @throws ORMException
+     */
+    public function downloadAction(
+        Request $request,
+        CoreParametersHelper $parametersHelper,
+        AssetModel $model,
+        string $slug,
+    ): Response {
+        try {
+            $entity = $model->getRepository()->findByIdAndAlias($slug);
+        } catch (NonUniqueResultException|EntityNotFoundException|\InvalidArgumentException) {
+            return $this->notFound();
+        }
+
+        return $this->createAssetResponse($request, $parametersHelper, $model, $entity);
```

Comment:
Nitpick, but this would read better:
```suggestion
        try {
            $entity = $model->getRepository()->findByIdAndAlias($slug);

            return $this->createAssetResponse($request, $parametersHelper, $model, $entity);
        } catch (NonUniqueResultException|EntityNotFoundException|\InvalidArgumentException) {
            return $this->notFound();
        }
```